base_location/tests/test_base_location.py

```diff
@@ -231,6 +231,66 @@ def test_display_name(self):
             ).name
         )
 
+    def test_name_search___can_find_using_city_name_or_zip_or_code(self):
+        barcelona_data = {
```

Comment:
Don't create again this cities, but modify setUp to include missing fields like code

Reply:
I'm not creating the cities, I'm creating res.better.zip records in exactly the same way as the other tests are doing through `values_better_zip1` and `values_better_zip2`, reusing all the available data (and relative records) from the cities created in `setUp` and adding the missing pieces of information which are needed only for this test. 

If this approach is wrong/difficult to understand (which I do strongly agree with), the entire test suite for the module could be refactored to use a common set of data and records created in `setUp`, but that should be out of the scope of this PR.

Reply:
No, I mean to move that to the setUp. It's not mandatory. It's only a question of readibility. OK, let's keep it this way right now.

Reply:
Cool, thank you very much for the quick review!